content/docs/user-guide/data-management/data-versioning.md

```diff
@@ -41,3 +41,33 @@ To simplify remote data operations, DVC now supports native versioning of files
 and directories on several cloud providers. This means that you can browse your
 files normally as you would see them in your local workspace.
 -->
+
+## Project configuration
+
+Besides metafiles, <abbr>DVC projects</abbr> may contain a config file
+(`.dvc/config`) that can also be treated as code when it comes to version
+control.
+
+<admon icon="book">
+
+See `dvc config` for more information on DVC config.
+
+</admon>
+
+Some times it's important to version configuration changes along with
+corresponding data updates. Most notably, if you [set up remote storage] and
+`dvc push` data for others to `dvc pull` later, you should `git commit` both the
+metafile(s) and `.dvc/config` to the repo.
+
+Advanced situations where this is may also be necessary:
+
+- When migrating to a [shared cache]
+- If you enable `dvc cache hydra` and then create an [experiment queue] with
+  [param composition] to run from another environment
```

Comment:
Added this info (new section) to the Data Versioning guide.

Reply:
> Advanced situations

p.s. any other scenarios like this? I went through all the config sections and couldn't think of anything else.